manta-crypto/src/merkle_tree/tree.rs

```diff
@@ -692,16 +699,25 @@ where
     }
 }
 
-impl<C, COM> accumulator::Model<COM> for Parameters<C, COM>
+impl<C, COM> accumulator::Types for Parameters<C, COM>
 where
     C: Configuration<COM> + ?Sized,
-    COM: Has<bool>,
+    COM: Has<bool> + NonNative,
     InnerDigest<C, COM>: ConditionalSwap<COM> + constraint::PartialEq<InnerDigest<C, COM>, COM>,
     LeafDigest<C, COM>: ConditionalSwap<COM>,
 {
     type Item = Leaf<C, COM>;
     type Witness = PathVar<C, COM>;
     type Output = Root<C, COM>;
+}
+
+impl<C, COM> accumulator::Model<COM> for Parameters<C, COM>
+where
+    C: Configuration<COM> + ?Sized,
+    COM: Has<bool> + NonNative,
+    InnerDigest<C, COM>: ConditionalSwap<COM> + constraint::PartialEq<InnerDigest<C, COM>, COM>,
+    LeafDigest<C, COM>: ConditionalSwap<COM>,
+{
```

Comment:
About the NonNative trait: is the idea to do a general splitting everywhere of those implementations that depend on COM into implementations that don't and implementations that depend on NonNative COM (i.e. is the ultimate goal to generalize the splitting you've done when implementing accumulator::Model for Parameters?). I see why that might be interesting (code for () can be much simpler than code for a generic compiler COM), but in the particular case you apply the splitting it doesn't seem so, as both implementations look almost identical to each other and also to the general code for any COM that existed before.

Reply:
The `NonNative` trait used to be "implemented" by the `Has<bool>` trait when `()` did not implement it. It now does, so we cannot use that as the way of distinguishing Rust from non-Rust compilers. Instead, the `NonNative` trait allows us to explicitly distinguish them so we can give different implementations for each case. For merkle tree path verification, we actually have two different data structures `Path` and `PathVar` which are not the same and so we need two different verification functions.